### PR TITLE
document: steady-state

### PR DESCRIPTION
**Pull request by joshuaroot**
Regarding #6 
Introducing a document to help explain `steady-state`

### CODE REVIEW

**Comment by russmiles** [comment]
Let me know when this is ready to review @joshuaroot

**Comment by russmiles** [comment]
Also, should steady state be hyphenated? I often use Steady-State Hypothesis, but it does seem to deviate in definitions from different sources...

**Comment by joshuaroot** [comment]
Honestly, I've look up if it should be hyphenated... and I get yes, unless you don't want to, then no... 
`steady` is an adjective, `state` is a noun... because we're using `steady-state` as an adjective to modify the noun `hypothesis`. I believe the use of a hyphen is allowed via the compound modifier.

File: chaos-engineering-steady-state.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+**Chaos Engineering Fundamentals: The Steady State**
+
+In chaos engineering, the Steady State is the core idea surrounding each experiment. Without this, running disruptions against resources is actually chaos, instead of controlled experiments. 
+
+**What is Steady State?**
+
+The Steady State defines what you believe to be the status of your environment before and after a chaos experiment runs. Any change to this hypothesis results in a deviation (a result) that can be investigated and corrected for. In other words, if your system does not return to its expected Steady State after running a chaos experiment, that should be a red flag that you need to investigate your system’s resilience.
```
Comment (russmiles) [inline chaos-engineering-steady-state.md]: Before, after and perhaps even during?

Comment (russmiles) [inline chaos-engineering-steady-state.md]: "that is a candidate for investigation and potentially a place for improvement to be applied."?

Comment (russmiles) [inline chaos-engineering-steady-state.md]: system's robustness strategies? Instead of resilience?

File: chaos-engineering-steady-state.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+**How do I declare my Steady State Hypothesis?**
+
+You can determine which deviations from Steady State get flagged by defining the tolerance range. The probe tolerance can be defined by a number of factors based on the probe itself. The most rigid tolerances check for a boolean or count value, while more flexible probes check for a value in a json path or regular expression. The last type of tolerance can be defined using another probe. These tolerances should be used carefully as they can cause an experiment to fail if not defined correctly.
+
+It is recommend that you make your steady-state-hypothesis as detailed as possible to get the most accurate results. To define your Steady State using Chaos Toolkit, for example, you need to use probes that scan your environment before and after the experiment runs. Below is an example snippet for defining a probe.
```
Comment (russmiles) [inline chaos-engineering-steady-state.md]: It is recommended

File: chaos-engineering-steady-state.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+       }
+     }
+   ]
+ }
+}
+```
+
+Let’s break down what’s happening in this hypothesis:
+
+- First, we have to define what type of probe we are going to use. For all probes related to checking the value of a cloud resource, the type will (generally) be “python”. For probes that will interact internally to a resource (ex: a path exists on an instance or a specific process is running) the type will vary. We also give a “name” to the probe, this is just a way to easily define what the probe will do.
```
Comment (russmiles) [inline chaos-engineering-steady-state.md:57]: Maybe add links to the Chaos Toolkit docs for concepts such as probes?

File: chaos-engineering-steady-state.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+- Next, we are declaring our tolerance. Both the probes above return a boolean value. The first probe “asg is healthy” is checking to see that the “desired” number of instances matches the number of “healthy” instances in the ASG. The second probe is validating that the ASG is not currently scaling up or down.
+
+If both of these probes are true, the steady-state passes and the experiment can begin. If one or both of the probes are not true, the experiment fails before moving onto the actions.
+
+If a probe fails after the experiment, this provides a useful result that you can research to find the cause of the deviation. Once you have corrected for this deviation, run the experiment again to prove your correction works. 
+
+As you can see, Steady State analysis becomes most useful when it is run frequently and as part of the experiment. Knowing your Steady State is the key to successful and meaningful chaos engineering implementation.
```
Comment (russmiles) [inline chaos-engineering-steady-state.md]: key to a successful and meaningful chaos engineering experiment implementation?

**Review by russmiles — CHANGES_REQUESTED** [review]
Loved it, just some possible changes to apply at your discretion.

**Review by russmiles — APPROVED** [review]
LGreatTM

**Pull request merged**